sympy/matrices/expressions/matexpr.py

```diff
@@ -1026,6 +1026,66 @@ def _entry(self, i, j, **kwargs):
         return S.One
 
 
+class MatrixWild(MatrixSymbol, Wild):
+    """A wildcard for matrix expressions based on ``Wild``.
+
+    Examples
+    ========
+
+    >>> from sympy import MatrixSymbol
+    >>> from sympy.matrices.expressions.matexpr import MatrixWild
+    >>> A, B = MatrixSymbol('A', 3, 3), MatrixSymbol('B', 3, 3)
+    >>> W = MatrixWild('W', 3, 3)
+    >>> (A*B).match(W)
+    {W_: A*B}
+    >>> (A*B).match(A*W)
+    {W_: B}
+    >>> (A**2).match(A*W)
+    {W_: A}
```

Comment:
Maybe also add an example

```
>>> print(A.match(A*W))
None
```

to show that it returns None when there are no matches

It would also be good to show exclude and properties in an example as well.